pkgs/development/tools/azure-functions-core-tools/default.nix

```diff
@@ -1,94 +1,45 @@
-{
-  stdenv,
-  lib,
-  config,
-  fetchurl,
-  unzip,
-  makeWrapper,
-  icu,
-  libunwind,
-  curl,
-  zlib,
-  libuuid,
-  openssl,
-}: let
-  platforms = {
-    "aarch64-darwin" = {
-      platformStr = "osx-arm64";
-      hash = "sha256-yp3VTt5m8KuACjrBIotfQ5ZdgMvfwYIFaqY2475pHRs=";
-    };
-    "x86_64-darwin" = {
-      platformStr = "osx-x64";
-      hash = "sha256-bTHh0mwGbe6JVsR8rDHGpGJ2+AipHb8NIBIW7iiuz6I=";
-    };
-    "x86_64-linux" = {
-      platformStr = "linux-x64";
-      hash = "sha256-5R4/hCxCz6KfBl9Zbei+iFty5S2MOYt9hMvPMjCzL54=";
-    };
+{ lib,
+  callPackage,
+  fetchFromGitHub,
+  buildDotnetModule,
+  buildGoModule,
+  dotnetCorePackages,
+}:
+let
+  version = "4.0.5390";
+  src = fetchFromGitHub {
+    owner = "Azure";
+    repo = "azure-functions-core-tools";
+    rev = "${version}";
+    sha256 = "sha256-pbhtixJh585gHTJeqtYedQzC5/LL/Pzu5JlC9cWVW18=";
   };
-
-  platformInfo = builtins.getAttr stdenv.targetPlatform.system platforms;
+  gozip = callPackage ./gozip.nix { inherit lib fetchFromGitHub buildGoModule src version; };
```

Comment:
you only need to pass `src` and `version` - the rest of the arguments will be filled out by `callPackage`